hamilton/function_modifiers/expanders.py

```diff
@@ -869,6 +874,156 @@ def extractor_fn(
         return output_nodes
 
 
+def _process_unpack_fields(fields: List[str], output_type: Any) -> Optional[List[Type]]:
+    """Processes the fields and base output type to extract a tuple of field types.
+
+    :param fields: List of fields to extract from the tuple.
+    :param output_type: The output type of the node function.
+    :return: List of field types.
+    """
+
+    base_type = typing_inspect.get_origin(output_type)
+    if base_type != tuple and base_type != Tuple:
+        return
+
+    output_args = typing_inspect.get_args(output_type)
+    num_ellipsis = output_args.count(Ellipsis)
+    if num_ellipsis > 1:
+        raise base.InvalidDecoratorException(
+            f"Invalid tuple: Found more than one ellipsis ('...'): {output_type}"
+        )
+    elif num_ellipsis == 1:
```

Comment:
What happens if it is `tuple[..., int]` (which, addmittedly, isn't really logical...)

Reply:
Ah yes, good catch, I will add that!

I was actually trying to include some more tests for cases like `tuple[...]`, `tuple[int, int, ...]` but I was having some issues with backward compatibility. I will take another crack at it.